cli/providers/aws/translate.go

```diff
@@ -129,54 +130,54 @@ func loadStateless(t *gocf.Template, s *service.Service, e *environment.Environm
 }
 
 func loadASG(t *gocf.Template, s *service.Service, e *environment.Environment, ami, vpc, encodedCompose string) error {
-	instanceIngressRules := gocf.EC2SecurityGroupRuleList{}
-	loadbalancerIngressRules := gocf.EC2SecurityGroupRuleList{}
-	for _, port := range s.GetPorts() {
-		instancePortNumber, _ := strconv.ParseInt(port.InstancePort, 10, 64)
-		instanceIngressRules = append(
-			instanceIngressRules,
-			gocf.EC2SecurityGroupRule{
-				SourceSecurityGroupIdXXSecurityGroupIngressXOnlyX: gocf.Ref("ELBSecurityGroup").String(),
-				IpProtocol: gocf.String("tcp"),
-				FromPort:   gocf.Integer(instancePortNumber),
-				ToPort:     gocf.Integer(instancePortNumber),
-			})
-		portNumber, _ := strconv.ParseInt(port.Port, 10, 64)
-		loadbalancerIngressRules = append(
-			loadbalancerIngressRules,
-			gocf.EC2SecurityGroupRule{
-				CidrIp:     gocf.String("0.0.0.0/0"),
-				IpProtocol: gocf.String("tcp"),
-				FromPort:   gocf.Integer(portNumber),
-				ToPort:     gocf.Integer(portNumber),
-			})
-	}
 	securityGroups := []gocf.Stringable{gocf.String(e.Provider.SecurityGroup)}
-	if len(instanceIngressRules) > 0 {
-		securityGroup := &gocf.EC2SecurityGroup{
-			GroupDescription:     gocf.String(fmt.Sprintf("Instance Security Group for %s", s.GetFullName(e, "-"))),
-			SecurityGroupIngress: &instanceIngressRules,
-			VpcId:                gocf.String(vpc),
+	loadBalancerNames := gocf.StringList()
+	if s.Public {
+		instanceIngressRules := gocf.EC2SecurityGroupRuleList{}
+		loadbalancerIngressRules := gocf.EC2SecurityGroupRuleList{}
+		for _, port := range s.GetPorts() {
+			instancePortNumber, _ := strconv.ParseInt(port.InstancePort, 10, 64)
```

Comment:
maybe we should make port.InstancePort a int, so that we get the validation for free on the deserialization, and we don't need to do this type of parsing later.

Reply:
It is a string in other many places, not sure which one would be better, but I think it is out of the scope of this PR